flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala

```diff
@@ -585,6 +585,50 @@ object FlinkRexUtil {
     }
     false
   }
+
+  /**
+   * Returns whether a given expression is deterministic in streaming scenario, differs from
+   * calcite's [[RexUtil]], it considers both non-deterministic and dynamic functions.
+   */
+  def isDeterministicInStreaming(e: RexNode): Boolean = try {
```

Comment:
Why we emphasize `streaming` ?

we should add test cases for this util method

Reply:
I want to highlight it to avoid some one using it in batch processing due to the different semantics on dynamic function.
I'll add separate tests for it.